src/libstore/remote-store.hh

```diff
@@ -62,9 +62,9 @@ public:
     void addToStore(const ValidPathInfo & info, Source & nar,
         RepairFlag repair, CheckSigsFlag checkSigs) override;
 
-    StorePath addToStore(const string & name, const Path & srcPath,
+    StorePath addToStoreFromDump(Source & dump, const string & name,
         FileIngestionMethod method = FileIngestionMethod::Recursive, HashType hashAlgo = htSHA256,
-        PathFilter & filter = defaultPathFilter, RepairFlag repair = NoRepair) override;
+        RepairFlag repair = NoRepair) override;
```

Comment:
GitHub may render this as a "change" but it's really a deletion and an addition.

Addition:
 - Providing a real implementation instead of throwing an exception realizes the goal of this PR, allowing daemons to be chained

Deletion:
 - `addToStore(... Path ...)` has a default implementation that delegates to `addToStoreFromDump` which is all we need